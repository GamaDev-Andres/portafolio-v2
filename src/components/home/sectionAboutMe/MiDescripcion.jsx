import { useState } from 'react'
import { miDescripcion } from '../../../helpers/data'
import ArticleDescription from './ArticleDescription'

const MiDescripcion = () => {
  const [verMas, setVerMas] = useState(false)

  const handleVerMas = () => setVerMas(!verMas)
  return (
    <div className="text-fondos flex flex-col gap-4">
      <ArticleDescription obj={miDescripcion.quienSoy} title="Quien soy: " />
      {verMas && (
        <ArticleDescription obj={miDescripcion.proceso} title="proceso: " />
      )}
      <button onClick={handleVerMas} className=" mx-auto block text-blue-400">
        {verMas ? 'Ver menos' : 'Ver mas'}{' '}
        {verMas
          ? (
          <i className="fa-solid fa-arrow-up"></i>
            )
          : (
          <i className="fa-solid fa-arrow-down"></i>
            )}
      </button>
    </div>
  )
}

export default MiDescripcion
