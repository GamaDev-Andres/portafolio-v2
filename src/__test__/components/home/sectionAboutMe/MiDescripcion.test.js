/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import MiDescripcion from '../../../../components/home/sectionAboutMe/MiDescripcion'

describe('probando <MiDescripcion/>', () => {
  const wrapper = Enzyme.mount(
    <MiDescripcion />
  )
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('ArticleDescription').length).toBe(1)
    expect(wrapper.find('button').exists()).toBe(true)
  })
  test('debe montar otro articulo cuando haga click', () => {
    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.find('ArticleDescription').length).toBe(2)
  })
})
