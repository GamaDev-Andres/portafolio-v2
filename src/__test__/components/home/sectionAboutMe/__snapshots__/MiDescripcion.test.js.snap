// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <MiDescripcion/> debe montarse correctamente 1`] = `
<div
  className="text-fondos flex flex-col gap-4"
>
  <article
    className="flex flex-col gap-2"
  >
    <p>
      <span
        className="text-blue-400 font-medium"
      >
        Quien soy: 
      </span>
      Mi nombre es Andres Gama, me apasiona encontrar grandes retos que me
          permitan crecer profesional y personalmente, la programación me
          permite enfocarme en esto.Actualmente tengo 21 años y sé que lograre
          que mis conocimientos sean participes de grandes, útiles y reconocidos
          proyectos.
    </p>
    <p>
      En mí podras ver a un chico ansioso por aprender y dispuesto a
          afrontar complejas curvas de aprendizaje y gracias a mi constancia y
          disciplina veras a alguien creciendo constantemente.
    </p>
  </article>
  <button
    className=" mx-auto block text-blue-400"
    onClick={[Function]}
  >
    Ver mas
     
    <i
      className="fa-solid fa-arrow-down"
    />
  </button>
</div>
`;
